frontend/packages/dev-console/src/components/pipelines/utils/triggers.ts

```diff
@@ -83,11 +76,11 @@ export type RouteTemplate = {
   triggerTemplateName: string;
 };
 
-export const usePipelineTriggerTemplateNames = (pipeline: Pipeline): RouteTemplate[] | null => {
-  const eventListenerResources = useAllEventListeners(pipeline);
-  const {
-    metadata: { namespace },
-  } = pipeline;
+export const usePipelineTriggerTemplateNames = (
+  pipelineName: string,
+  namespace: string,
+): RouteTemplate[] | null => {
+  const eventListenerResources = useAllEventListeners(namespace);
```

Comment:
Why did you make this change? As far as I can tell you possessed the Pipeline object in all cases where you used this function. These changes are undesirable and lead to coupling of functionality. 

Prior to this change, the caller just knew they had a Pipeline and needed to get related objects of type `X`. `usePipelineTriggerTemplates` makes a lot less sense when you're passing a name and a namespace, rather than the whole Pipeline.

I disagree with this change, and unless you have a solid reason I can't see, I recommend undoing this.

Reply:
I make this change so that I can use this ```usePipelineTriggerTemplateNames``` hook in ```PipelineDetailsPage``` as the whole pipeline object is not present on pipeline details age.

Reply:
Really? 🤔 That seems odd.

Reply:
Yes, this is the only reason to make this change.